src/Chapter3.hs

```diff
@@ -374,6 +381,29 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+type Coins = Int
+
+data Knight = Knight
+    { knightHealth :: Float
+    , knightAttack :: Float
+    , knightGold   :: Coins
+    } deriving (Show)
+
+data Monster = Monster
+    { monsterHealth :: Float
+    , monsterAttack :: Float
+    , monsterGold   :: Coins
+    } deriving (Show)
+
+fight :: Knight -> Monster -> Coins
+fight knight monster
+      | kinghtSteps < monsterSteps = knightGold knight + monsterGold monster
+      | kinghtSteps > monsterSteps = -1
+      | otherwise = knightGold knight
+    where
+      kinghtSteps = monsterHealth monster / knightAttack knight
+      monsterSteps = knightHealth knight / monsterAttack monster
```

Comment:
Interesting decision! ⚔️ 
Could you please elaborate a bit on what the `step` is in this context? 🙂

Reply:
As I understand knight and monster should hit each other one by one in turn until one of them will die. 
So I need to know how much rounds (steps) need for each of them. The one with fewer steps will win ;)

Reply:
Oh, very clever! I see now! 👍🏼